autocomplete/bash_autocomplete

```diff
@@ -3,14 +3,19 @@
 : ${PROG:=$(basename ${BASH_SOURCE})}
 
 _cli_bash_autocomplete() {
+  if [[ "${COMP_WORDS[@]:0:$COMP_CWORD}" != "source" ]]; then
     local cur opts base
     COMPREPLY=()
     cur="${COMP_WORDS[COMP_CWORD]}"
-    opts=$( ${COMP_WORDS[@]:0:$COMP_CWORD} --generate-bash-completion )
+    if [[ $cur == -* ]]; then
+      opts=$( ${COMP_WORDS[@]:0:$COMP_CWORD} ${cur} --generate-bash-completion )
+    else
+      opts=$( ${COMP_WORDS[@]:0:$COMP_CWORD} --generate-bash-completion )
+    fi
     COMPREPLY=( $(compgen -W "${opts}" -- ${cur}) )
     return 0
+  fi
 }
 
-complete -F _cli_bash_autocomplete $PROG
-
+complete -o bashdefault -o default -o nospace -F _cli_bash_autocomplete $PROG
```

Comment:
I don't 100% follow everything that's going on here - but I think we can definitely merge and iterate on bug fixes 🔧

Reply:
Sourcing this file causes bash builtin `source` command to throw `-bash: source: --: invalid option` error on <kbd>TAB</kbd> . I was not able detect why in the first place `source` builtin is getting linked to this function. So excluded it and fallback to defaults